<?xml version="1.0"?>
<launch>
  <!-- Run the map server -->
  <arg name="map_file" default="$(find navigation)/map/dt_map.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

  <!-- Run the odom -> map transform publisher -->
  <node name="odom_to_map_node" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 0 0 0 map odom" />

  <!-- Create map from static ground truth-->
  <arg name="no_static_map" default="false"/>

  <!-- Set global and local planners-->
  <arg name="base_global_planner" default="global_planner/GlobalPlanner"/>
  <arg name="base_local_planner" default="teb_local_planner/TebLocalPlannerROS"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  
    <rosparam file="$(find navigation)/config/navigation_nodes/my_move_base.yaml" command="load"/>

    <!-- observation sources located in costmap_common.yaml -->
    <rosparam file="$(find navigation)/config/navigation_nodes/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find navigation)/config/navigation_nodes/costmap_common.yaml" command="load" ns="local_costmap" />

    <!-- local costmap, needs size -->
    <rosparam file="$(find navigation)/config/navigation_nodes/local_costmap.yaml" command="load" ns="local_costmap" />

    <!-- static global costmap, static map provides size -->
    <rosparam file="$(find navigation)/config/navigation_nodes/global_costmap.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/>
  </node>
</launch>
